src/drivers/intel/ssp/ssp.c

```diff
@@ -765,11 +802,13 @@ static void ssp_start(struct dai *dai, int direction)
 
 	spin_lock(&dai->lock);
 
-	/* request mclk/bclk */
-	ssp_pre_start(dai);
+	if (!ssp->params.clk_early_start) {
+		/* request mclk/bclk */
+		ssp_pre_start(dai);
 
-	/* enable port */
-	ssp_update_bits(dai, SSCR0, SSCR0_SSE, SSCR0_SSE);
+		/* enable port */
+		ssp_update_bits(dai, SSCR0, SSCR0_SSE, SSCR0_SSE);
+	}
```

Comment:
I think we should enable MCLK and BCLK separately. They are not logically connected.